### ISSUE TITLE
Should avoid generating inline sourcemaps for unchanged *.js files

### ISSUE THREAD

**Issue by rwjblue** [ember-template-imports/ember-template-imports#108]
When `ember-template-imports` gets added **all** `.js` files end up including new inline sourcemaps. 

Quick summary:

1. ember-template-imports adds a custom preprocessor for `.js` files [here](https://github.com/ember-template-imports/ember-template-imports/blob/72f11d2f145bc20e1912339475ef17023fa4722f/ember-addon-main.js#L35-L38)
2. that preprocessor (from [here](https://github.com/ember-template-imports/ember-template-imports/blob/72f11d2f145bc20e1912339475ef17023fa4722f/src/preprocessor-plugin.js#L77-L94)) calls `preprocessEmbeddedTemplates` 
3. that uses `magic-string` to process [specific template-literals](https://github.com/ember-template-imports/ember-template-imports/blob/72f11d2f145bc20e1912339475ef17023fa4722f/src/preprocess-embedded-templates.ts#L229-L239)
4. it uses magic-string for all .js files, it ends up including [sourcemaps for every .js file](https://github.com/ember-template-imports/ember-template-imports/blob/72f11d2f145bc20e1912339475ef17023fa4722f/src/preprocess-embedded-templates.ts#L257-L268) regardless if there were any templates in it

---

Tldr; we should change 

https://github.com/ember-template-imports/ember-template-imports/blob/72f11d2f145bc20e1912339475ef17023fa4722f/src/preprocess-embedded-templates.ts#L257-L268

to something like:

```js
const hasChanges = template !== output;
if (includeSourceMaps && hasChanges) {
   const { dir, name } = path.parse(relativePath); 
  
   const map = s.generateMap({ 
     file: `${dir}/${name}.js`, 
     source: relativePath, 
     includeContent: true, 
     hires: true, 
   }); 
  
   output += `\n//# sourceMappingURL=${map.toUrl()}`; 
}
```
